tokio-uds/src/frame.rs

```diff
@@ -0,0 +1,155 @@
+use std::io;
+use std::os::unix::net::SocketAddr;
+use std::path::PathBuf;
+
+use futures::{Async, Poll, Stream, Sink, StartSend, AsyncSink};
+
+use super::UnixDatagram;
+
+use tokio_codec::{Decoder, Encoder};
+use bytes::{BytesMut, BufMut};
+
+/// A unified `Stream` and `Sink` interface to an underlying `UnixDatagram`, using
+/// the `Encoder` and `Decoder` traits to encode and decode frames.
+///
+/// Unix datagram sockets work with datagrams, but higher-level code may wants to
+/// batch these into meaningful chunks, called "frames". This method layers
+/// framing on top of this socket by using the `Encoder` and `Decoder` traits to
+/// handle encoding and decoding of messages frames. Note that the incoming and
+/// outgoing frame types may be distinct.
+///
+/// This function returns a *single* object that is both `Stream` and `Sink`;
+/// grouping this into a single object is often useful for layering things which
+/// require both read and write access to the underlying object.
+///
+/// If you want to work more directly with the streams and sink, consider
+/// calling `split` on the `UnixDatagramFramed` returned by this method, which will break
+/// them into separate objects, allowing them to interact more easily.
+#[must_use = "sinks do nothing unless polled"]
+#[derive(Debug)]
+pub struct UnixDatagramFramed<C> {
```

Comment:
Not to bike shed, but could this be called `UdsFramed`? Mostly because [UDP](https://github.com/tokio-rs/tokio/blob/master/tokio-udp/src/frame.rs#L29) uses the acronym.

Reply:
I was just trying to match the base `UnixDatagram` name. Would you prefer it match that or match the udp's framed name?

Reply:
Yes, you are correct. I meant to delete the comment before hitting submit. The name is fine.